Victor-Ugwueze/MyDiary
# #159310223,#159310180 feature auth signup

### What does this PR do ?
This PR has will adds 
   * user signup
   * user login
   * user authentication with jsonwebtoken
   * testing

### Description of task to be completed ?
Implement the following:
  * API endpoints
      1.  `/auth/signup`
      2.  `/auth/login`
  * tests for the above endpoints
  * Authenticate user with jwt
   * Add the following files
     server/helpers/table_schemas/dropTables.js
      server/helpers/test_data/seed_database.js
   
       
### How should this be tested
You should have PostgreSQL installed before you get started.  Then clone this repo, run `npm install` to install all dependencies and devDependencis. 
Open server/helpers/dbHelper.js change the connection string to match that of your local machine.
Then run `npm run start-dev`

To run test, run `npm test` Note this test requires database

### What are the relevant pivotal tracker stories?
#159310223
#159310180
